src/configurationProvider.ts

```diff
@@ -0,0 +1,74 @@
+import * as vscode from 'vscode';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { parse } from 'jsonc-parser';
+import { OmniSharpServer } from './omnisharp/server';
+import * as serverUtils from './omnisharp/utils';
+import { AssetGenerator, addTasksJsonIfNecessary, createLaunchConfiguration, createAttachConfiguration, containsDotNetCoreProjects} from './assets';
+import { WorkspaceInformationResponse } from './omnisharp/protocol';
+
+export class CSharpConfigurationProvider implements vscode.DebugConfigurationProvider {
+    private server: OmniSharpServer;
+
+    public constructor(server: OmniSharpServer) {
+        this.server = server;
+    }
+
+    /**
+     * TODO: Remove function. 
+     * 
+     * Note: serverUtils.requestWorkspaceInformation only retrieves one project. Therefore, generator will be incorrect for all folders
```

Comment:
> serverUtils.requestWorkspaceInformation only retrieves one project.

This isn't true.

Reply:
Is there a way to make it retrieve all projects in the current workspace for VsCode?

Reply:
It will retrieve all projects loaded by OmniSharp. Currently, OmniSharp can only target a single folder. However, that folder contains multiple `.csproj` projects or an `.sln` with multiple projects, it will return them. We're going to be doing some work in omnisharp-roslyn to make OmniSharp support multiple folders: https://github.com/OmniSharp/omnisharp-roslyn/issues/909